kubespawner/spawner.py

```diff
@@ -2393,7 +2393,16 @@ def on_reflector_failure():
                 on_failure=on_reflector_failure,
                 **kwargs,
             )
-            asyncio.ensure_future(self.__class__.reflectors[key].start())
+            f = asyncio.ensure_future(self.__class__.reflectors[key].start())
+
+            async def catch_reflector_start():
+                try:
+                    await f
+                except Exception as e:
+                    self.log.exception(f"Reflector for {kind} failed to start.")
+                    sys.exit(1)
```

Comment:
I'm not feeling confident on the outcome of having `sys.exit(1)` here yet.

I looked for `sys.exit` within the source code to see if its not the first time this is done, and therefore probably not an issue. Doing that I found this commit which added a reason for not feeling confident that sys.exit should be called: https://github.com/jupyterhub/kubespawner/commit/c15ad869b6c60a79dfa0f727e429ab4750cf84ff

Reply:
This is identical to the call directly above which is called for the same reason, but when the error occurs later rather than at start time. We know it works because this is the call that's restarting unhealthy 2i2c Hubs (working as intended).

It was more complicated when we had threads, but we don't anymore. Sys.exit from the main thread always unconditionally exits the process unless unless SystemExit is caught explicitly.

Reply:
Ah, thanks!